test/unit/components/DataCollector/index.spec.js

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import createDataCollector from "../../../../src/components/DataCollector/index";
+import createPayload from "../../../../src/core/network/createPayload";
+import { defer } from "../../../../src/utils";
+
+const flushPromises = returnValue => {
+  const deferred = defer();
+  setTimeout(() => deferred.resolve(returnValue), 0);
+  return deferred.promise;
+};
+
+describe("Event Command", () => {
+  let eventCommand;
+  let onBeforeEventSpy;
+  let onBeforeDataCollectionSpy;
+  let sendRequestSpy;
+  const lifecycle = {
+    onBeforeEvent: () => Promise.resolve(),
+    onBeforeDataCollection: () => Promise.resolve()
+  };
+  const network = {
+    createPayload,
+    sendRequest: () => flushPromises({})
+  };
+  beforeEach(() => {
+    onBeforeEventSpy = spyOn(lifecycle, "onBeforeEvent").and.callThrough();
+    onBeforeDataCollectionSpy = spyOn(
+      lifecycle,
+      "onBeforeDataCollection"
+    ).and.callThrough();
+    sendRequestSpy = spyOn(network, "sendRequest").and.callThrough();
+    const dataCollector = createDataCollector();
+    dataCollector.lifecycle.onComponentsRegistered({ lifecycle, network });
+    eventCommand = dataCollector.commands.event;
+  });
+  afterEach(() => {
+    return flushPromises();
+  })
+
+  it("Calls onBeforeEvent", () => {
+    return eventCommand({}).then(() => {
+      expect(lifecycle.onBeforeEvent).toHaveBeenCalledWith(
+        jasmine.anything(),
+        false
+      );
+    });
+  });
+  it("Extracts isViewStart for onBeforeEvent", () => {
+    return eventCommand({ type: "viewStart" }).then(() => {
+      expect(lifecycle.onBeforeEvent).toHaveBeenCalledWith(
+        jasmine.anything(),
+        true
+      );
+    });
+  });
+  it("Calls onBeforeEvent with a matching event", () => {
+    eventCommand({ data: { a: 1 }, meta: { b: 2 } }).then(() => {
+      expect(lifecycle.onBeforeEvent.calls.argsFor(0)[0].toJSON().data).toEqual(
+        { a: 1 }
+      );
+      expect(lifecycle.onBeforeEvent.calls.argsFor(0)[0].toJSON().meta).toEqual(
+        { b: 2 }
+      );
```

Comment:
It might be better to do:

```
expect(lifecycle.onBeforeEvent).toHaveBeenCalledWith({
  data: {
    a: 1
  },
  meta: {
    b: 2
  }
});
```